src/devices/cpu/z80/z80.cpp

```diff
@@ -3320,11 +3320,24 @@ void z80_device::take_interrupt()
 					PCD = irq_vector & 0xffff;
 					break;
 				default:        /* rst (or other opcodes?) */
-					/* RST $xx cycles */
-					CC(op, 0xff);
-					T(m_icount_executing - MTM * 2);
-					wm16_sp(m_pc);
-					PCD = irq_vector & 0x0038;
+					if (irq_vector == 0xfb)
+					{
+						// EI
+						CC(op, 0xfb);
+						T(m_icount_executing);
+						ei();
+					}
+					else if ((irq_vector & 0xc7) == 0xc7)
```

Comment:
This was added so that it won't silently execute the RST instruction, if it is not an RST instruction, so if anyone else runs into this, there will be a clear error log shown.